sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -392,6 +394,14 @@ case class HashAggregateExec(
      """.stripMargin
   }
 
+  private def typeNameForCodegen(clazz: Class[_]): String = {
```

Comment:
It might be better to move this helper function to `CodeGenerator`.

Reply:
and just call `typeName`?

Reply:
Nice suggestions of both comments! Will address.